src/app/modules/ix-forms/components/ix-combobox/ix-combobox.component.ts

```diff
@@ -116,21 +118,25 @@ export class IxComboboxComponent implements ControlValueAccessor, OnInit {
         if (setOption) {
           this.selectedOption = setOption ? { ...setOption } : null;
           if (this.selectedOption) {
-            this.filterChanged$.next('');
+            this.filterChanged$.next({ value: '' });
           }
         } else {
           /**
            * We are adding a custom fake option here so we can show the current value of the control even
            * if we haven't found the correct option in the list of options fetched so far. The assumption
            * is that the correct option exists in one of the following pages of list of options
            */
-          this.options.push({ label: this.value as string, value: this.value });
-          this.selectedOption = { ...this.options.find((option: Option) => option.value === this.value) };
-          if (this.selectedOption) {
-            this.filterChanged$.next('');
+          if (!isCalledOnInit) {
```

Comment:
Why not just `if(!this.allowCustomValue) {`? It is much easier and the end result seems like what we need.